website/docs/r/dl_route_report.html.markdown

```diff
@@ -0,0 +1,112 @@
+---
+subcategory: "Direct Link Gateway"
+layout: "ibm"
+page_title: "IBM : dl_route_report"
+description: |-
+  Manages IBM Direct Link Gateway.
+---
+
+# ibm_dl_route_report
+
+Create a Direct Link Route Report by using the Direct Link Route Report resource. For more information, see [about Direct Link Route Report](https://cloud.ibm.com/docs/dl?topic=dl-generate-route-reports&interface=ui).
+
+
+## Example usage to create Direct Link Route Report on dedicated gateway
+In the following example, you can create Direct Link Route Report for dedicated gateway:
+
+```terraform
+data "ibm_dl_routers" "test_dl_routers" {
+		offering_type = "dedicated"
+		location_name = "dal10"
+	}
+
+resource ibm_dl_gateway test_dl_gateway {
+  bgp_asn =  64999
+  global = true 
+  metered = false
+  name = "Gateway1"
+  speed_mbps = 1000 
+  type =  "dedicated" 
+  cross_connect_router = data.ibm_dl_routers.test_dl_routers.cross_connect_routers[0].router_name
+  location_name = data.ibm_dl_routers.test_dl_routers.location_name
+  customer_name = "Customer1" 
+  carrier_name = "Carrier1"
+
+} 
+
+resource ibm_dl_route_report dl_route_report {
+    gateway = ibm_dl_gateway.test_dl_gateway.id
+}
+```
+
+## Sample usage to create Direct Link Route Report on connect gateway
+In the following example, you can create Direct Link Route Report on connect gateway:
+
+
+```terraform
+data "ibm_dl_ports" "test_ds_dl_ports" {
+ 
+ }
+resource "ibm_dl_gateway" "test_dl_connect" {
+  bgp_asn =  64999
+  global = true
+  metered = false
+  name = "dl-connect-gw-1"
+  speed_mbps = 1000
+  type =  "connect"
+  port =  data.ibm_dl_ports.test_ds_dl_ports.ports[0].port_id
+}
+
+resource ibm_dl_route_report dl_route_report {
+    gateway = ibm_dl_gateway.test_dl_gateway.id
+}
+```
+
+## Argument reference
```

Comment:
Please make sure argument reference matches with resource schema.
Any Required / Optional / Optional and Computed arguments in the schema should be mentioned in Argument reference section

Any argument that is marked as only computed in schema should be mentioned in Attribute reference..

All